src/liballoc/rc.rs

```diff
@@ -1796,26 +1803,20 @@ impl<T> Weak<T> {
     /// [`forget`]: ../../std/mem/fn.forget.html
     #[stable(feature = "weak_into_raw", since = "1.45.0")]
     pub unsafe fn from_raw(ptr: *const T) -> Self {
-        if ptr.is_null() {
```

Comment:
Ah so this is from back when we used NULL as sentinel value for dangling `Weak`?

Reply:
Yep, exactly.